.github/workflows/test-canary.yml

```diff
@@ -0,0 +1,137 @@
+# This pipeline purpose is solely meant to run a subset of our test suites against upcoming or unreleased dependencies versions
+name: test-canary
+
+on:
+  push:
+    branches:
+      - main
+      - 'release/**'
+  pull_request:
+    paths-ignore:
+      - '**.md'
+
+env:
+  UBUNTU_VERSION: "24.04"
+
+  # Straight git branches
+  CONTAINERD_VERSION: "main"
+  # Windows install script can only set released versions
+  CONTAINERD_VERSION_WINDOWS: "2.0.0-rc.3"
+  IMGCRYPT_VERSION: "main"
+  BYPASS4NETNS_VERSION: "main"
+  KUBO_VERSION: "main"
+  GOTESTSUM_VERSION: "main"
+  # Set this to "next" for upcoming version (defined by `RegistryImageNext`)
+  DISTRIBUTION_VERSION: "next"
+  # FIXME. Currently pinned, see: https://github.com/containerd/nerdctl/pull/3153
+  RUNC_VERSION: "v1.1.13"
+
+  # Released versions - one can update these manually to override the Dockerfile pinned versions
+  BUILDKIT_VERSION: "v0.15.0-rc1"
```

Comment:
v0.15 is now GA

Reply:
See last comment. New version is now automatically retrieving the latest published release (including alpha/beta/RC).